app/controllers/AccountingPeriodController.scala

```diff
@@ -59,6 +62,34 @@ class AccountingPeriodController @Inject() (
     Ok(view(preparedForm, mode))
   }
 
+  private def updatedAnswersAndMode(request: OptionalDataRequest[AnyContent], form: AccountingPeriodForm, mode: Mode) =
+    Future.fromTry(request.userAnswers match {
+      case Some(answers) =>
+        answers
+          .get(AccountingPeriodPage)
+          .map {
+            case prevAnswer if prevAnswer == form =>
+              Success((answers, CheckMode))
+            case _ =>
+              answers.clean
+                .set(AccountingPeriodPage, form)
+                .map(_ -> NormalMode)
+          }
+          .getOrElse {
+            answers
+              .set(AccountingPeriodPage, form)
+              .map(_ -> mode)
+          }
+      case None =>
+        UserAnswers(request.userId)
+          .set(AccountingPeriodPage, form)
+          .map(_ -> mode)
+    })
+
+  private def accountingPeriodIsIrrelevant(form: AccountingPeriodForm) =
+    form.accountingPeriodStartDate.isBefore(LocalDate.parse("2022-04-02")) ||
+      form.accountingPeriodEndDate.get.isBefore(LocalDate.parse("2023-04-01"))
```

Comment:
I understand accountingPeriodEndDate will never be empty, but its a good practice to to avoid get on Option

```
form.accountingPeriodEndDate.exists(_.before(LocalDate.parse("2023-04-01"))
```

Reply:
Done